### ISSUE TITLE
Game crashes due to invalid UnitDestroyIndex in Build Anywhere variants

### ISSUE THREAD

**Issue by diptobi1** [kestasjk/webDiplomacy#586]
# Bug report

This is about a bug of Build Anywhere variants and the destroy code that might lead to crashes when a user did not enter valid destroy orders.

When a player has to destroy a unit but only provides invalid orders (e.g. by NMRing or by selecting no unit or the same unit twice), the UnitDestroyIndex is used to decide which unit should be destroyed. The index is calculated for each country, each territory and each unit type at variant installation and reflects the distance for an army or fleet in a certain territory from the nearest home SC. 
The index is only calculated for reachable territories (i.e. there is no index for fleets for in-land territories like Ruhr, Burgund, etc.). This is problematic for Build Anywhere variants as home SCs are not the only centers to build units on: It might be possible to get units on territories, that are not reachable from the home SCs. The adjudicator will then find no UnitDestroyIndex for the unit on the given territory and the game processor will crash. 
Take for example Moscow in the Classic Chaos variant. Due to Build Anywhere rules, Moscow might be able to get a fleet at some point. If Moscow's player then enters an invalid destroy order the adjudicator will not find a valid index for the fleet as no fleet on any territory will ever be reachable via a waterway by Moscow. The adjudicator will crash in this case.

## Expected behaviour

No crash. It is of course up to debate how the adjudicator should decide on destroys if the measure 'distance to home SCs' is no longer valid.

## Actual behaviour

The game crashes and needs an admin fixing the invalid orders by actually setting them correctly.

## Steps to reproduce

Take a test game of Classic Chaos. Give an inland country like Moscow, Warsaw, Vienna, Budapest, Serbia, Munich or Paris a fleet somewhere on the map. Take the home SC away so the fleet has to be destroyed. Do not enter orders this time and let the adjudicator decide to crash.

**Comment by jmo1121109** [comment]
Do you have a fix for this on vDip?

**Comment by diptobi1** [comment]
Yes, I added a code extension to all BuildAnywhere variants last year. Now random units on non-SC territories are selected for default destruction instead of using the Destroy-Index.

The corresponding commit is https://github.com/Sleepcap/vDiplomacy/commit/490b0fdf17f2d8845b6e3604bc334c13d4a491ee.

Would you like me to create pull request for webdip with this workaround?
